### PR TITLE
Information_schema case sensitiveness

### LINKED ISSUE #49510: MySQL compatibility: mixed case queries
**Use case**

When using MySQL protocol to connect BI tools (for example, QuickSight) to ClickHouse, the following query is executed to introspect the dataset size, and it fails:

```sql
SELECT 
  data_length 
FROM 
  information_schema.TABLES 
WHERE 
  table_schema = 'default' 
  AND table_name = 'cell_towers';
```

**Describe the solution you'd like**

Mixed case `<database>.<TABLE>` (or maybe even `<DATABASE>.<table>`) works.

**Additional context**

An example log of a failing query:

```
2023.04.25 13:26:44.562738 [ 319 ] {mysql:96:11ad169d-0139-456a-abc7-e9fdfae004f6} <Error> executeQuery: Code: 60. DB::Exception: Table information_schema.TABLES doesn't exist. (UNKNOWN_TABLE) (version 23.4.1.1157 (official build)) (from 35.158.127.201:43000) (in query: SELECT data_length FROM information_schema.TABLES WHERE table_schema = 'default' AND table_name = 'cell_towers')
```

### PR DESCRIPTION
**Pull request by yariks5s**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Closes #49510. Currently we have database and table names case-sensitive, but the tools query `information_schema` sometimes in lowercase, sometimes in uppercase.
For this reason we have `information_schema` database, containing lowercase tables, such as  `information_schema.tables` and `INFORMATION_SCHEMA` database, containing uppercase tables, such as `INFORMATION_SCHEMA.TABLES`.
But some tools are querying `INFORMATION_SCHEMA.tables` and `information_schema.TABLES`.
The proposed solution is to duplicate both lowercase and uppercase tables in lowercase and uppercase `information_schema` database.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-clickhouse** [comment]
This is an automated comment for commit da53bca6e4650b6d67737e6efd1b95232bf4ffc7 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/52695/da53bca6e4650b6d67737e6efd1b95232bf4ffc7/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: src/Storages/System/attachInformationSchemaTables.cpp [hunk 1]
```diff
@@ -41,6 +41,14 @@ static void createInformationSchemaView(ContextMutablePtr context, IDatabase & d
                                              database.getTableDataPath(ast_create), context, true).second;
 
         database.createTable(context, ast_create.getTable(), view, ast);
+        if (ast_create.getTable() == "tables")
```
Comment (alexey-milovidov) [inline src/Storages/System/attachInformationSchemaTables.cpp:44]: We need it for every table inside information_schema.

File: src/Storages/System/attachInformationSchemaTables.cpp [hunk 2]
```diff
@@ -41,13 +41,13 @@ static void createInformationSchemaView(ContextMutablePtr context, IDatabase & d
                                              database.getTableDataPath(ast_create), context, true).second;
 
         database.createTable(context, ast_create.getTable(), view, ast);
-        if (ast_create.getTable() == "tables")
+        if (database.getDatabaseName() == DatabaseCatalog::INFORMATION_SCHEMA)
```
Comment (alexey-milovidov) [inline src/Storages/System/attachInformationSchemaTables.cpp:44]: But here in the `createInformationSchemaView` it should be always about the information schema.

Comment (alexey-milovidov) [reply src/Storages/System/attachInformationSchemaTables.cpp:44]: So, we can create both variants unconditionally.

Comment (yariks5s) [reply src/Storages/System/attachInformationSchemaTables.cpp:44]: okay

**Review by KochetovNicolai — APPROVED** [review]

**Comment by KochetovNicolai** [comment]
test_tlsv1_3 is flaky

**Pull request merged**